packages/web-components/src/components/masthead/megamenu-top-nav-menu.ts

```diff
@@ -105,19 +105,33 @@ class DDSMegaMenuTopNavMenu extends DDSTopNavMenu {
         .querySelector('dds-masthead')
         ?.shadowRoot?.querySelector('.bx--masthead__l0');
 
+      // determine whether to apply margin-right on expand as HC has extra masthead styling
+      const cloudMasthead: HTMLElement | null | undefined = doc
+        .querySelector('dds-cloud-masthead-container')
+        ?.querySelector('dds-masthead')
+        ?.shadowRoot?.querySelector('.bx--masthead__l0');
+
       if (this.expanded) {
         doc.body.style.marginRight = `${this._scrollBarWidth}px`;
         doc.body.style.overflow = `hidden`;
         forEach(doc.querySelectorAll((this.constructor as typeof DDSMegaMenuTopNavMenu).selectorOverlay), item => {
           (item as DDSMegaMenuOverlay).active = this.expanded;
         });
-        if (masthead) {
+        if (cloudMasthead) {
+          if (doc.body.classList.contains('ibm-masthead-sticky') && doc.body.classList.contains('ibm-masthead-sticky-showing')) {
+            cloudMasthead.style.marginRight = `${this._scrollBarWidth}px`;
```

Comment:
@annawen1 this will be interesting to test across all browsers and their varying treatment of the scrollbar!

Reply:
haha we actually already have `margin-right` logic in our global masthead that works fine, just the cloud masthead needs adjustment because the `position` style was changed. Hoping it will work fine for them as well.